components/script/dom/event.rs

```diff
@@ -169,6 +170,9 @@ impl Event {
         // Step 3-4.
         let path = self.construct_event_path(&target);
         rooted_vec!(let event_path <- path.into_iter());
+        if self.type_() == atom!("click") {
+            is_activation_event = true;
```

Comment:
This variable isn't used, also you can fire manual click events that aren't activation events. Please remove the is_activation_event stuff.

Reply:
I think is_activition_event will be used in Steps 9.6.1 and Steps 9.8.1 of the [spec](https://dom.spec.whatwg.org/#concept-event-dispatch)

Reply:
Ah, I see. Can you implement that part?

Reply:
I can certainly try!

Reply:
Downcast it to an element, and if that succeeds, call `.as_maybe_activatable()` on it.